src/mpid/ch4/src/ch4r_win.h

```diff
@@ -159,9 +159,9 @@ static inline int MPIDIG_mpi_win_complete(MPIR_Win * win)
 
     /* FIXME: now we simply set per-target counters for PSCW, can it be optimized ? */
     do {
-        MPID_THREAD_CS_EXIT(VCI, MPIDI_global.vci_lock);
+        MPID_THREAD_CS_EXIT(VCI, MPIDI_VCI(MPIDI_VCI_ROOT).lock);
         MPIDIU_PROGRESS();
-        MPID_THREAD_CS_ENTER(VCI, MPIDI_global.vci_lock);
+        MPID_THREAD_CS_ENTER(VCI, MPIDI_VCI(MPIDI_VCI_ROOT).lock);
```

Comment:
Question: why do we do `CS_EXIT` around `MPIDIU_PROGRESS()`?

Reply:
It was introduced in this commit: https://github.com/pmodels/mpich/commit/1f506054dd54c3ae19b152fb05cde8de4fac2d90. I think we `CS_EXIT` to prevent recursive locking -- `MPIDIU_PROGRESS` pokes the progress engine, during which the VCI lock is taken again. @hajimefu could you verify?

Reply:
I see. The locks were taken in `src/mpid/ch4/src/ch4_win.h`. These `EXIT/ENTER` blocks are very weird code patterns, are there alternative ways? Can we call an unsafe version of `progress_test` here?

Reply:
@rzambre Yes I think that was the original intention. However, later @halimamer changed per-vci lock implementation to recursive one (https://github.com/pmodels/mpich/blob/master/src/mpid/common/thread/mpidu_thread_fallback.h#L152), so I'm not totally sure if this is still needed.

Reply:
Can't remember honestly why I left it there. One possible reason I left it is to avoid lock ordering deadlocks. `MPIDI_PROGRESS()` does global progress, so it will try to acquire other locks other than the `VCI` lock it holds, running to possible lock ordering issues.

My advice is to reinvestigate and drop this CS enter/exit code if you are certain that lock ordering issues will not happen.